shared/django_apps/rollouts/models.py

```diff
@@ -0,0 +1,50 @@
+from random import choice
+
+from django.contrib.postgres import fields
+from django.db import models
+
+
+def default_random_salt():
+    chars = []
+    ALPHABET = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    for _ in range(16):
+        chars.append(choice(ALPHABET))
+    return "".join(chars)
+
+
+class FeatureFlag(models.Model):
+    name = models.CharField(max_length=200, primary_key=True)
+    proportion = models.DecimalField(default=0, decimal_places=3, max_digits=3)
+    salt = models.CharField(max_length=10000, default=default_random_salt)
+
+    class Meta:
+        db_table = "feature_flags"
+        constraints = [
+            models.UniqueConstraint(fields=["name"], name="feature_flag_name")
+        ]
+        # TODO: assert that variant proportions sum to 1
```

Comment:
not sure if this is possible

Reply:
no big deal. i think if the variant proportions only add up to, say, 0.7, all that will happen is the remaining 0.3 will get the default value. nothing will blow up